ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -179,14 +205,38 @@ private QueryMetadata buildPlanForStructuredOutputNode(String sqlExpression, fin
               schemaKStream.getKeyField(),
               outputNode.getTimestampField(),
               outputNode.getKsqlTopic());
-
     }
 
-    if (updateMetastore) {
+    if (updateMetastore && outputNode.isDoCreateInto()) {
       metaStore.putSource(sinkDataSource.cloneWithTimeKeyColumns());
+    } else {
```

Comment:
This else includes all cases where updateMetastore is false (whether or not isDoCreateInto() returns true). Wouldn't this cause a null-pointer exception for cases where the sink data source doesn't exist yet (e.g. the ddl verification step for CSAS queries)